src/KubeOps.Cli/Generators/MutationWebhookGenerator.cs

```diff
@@ -22,9 +22,11 @@ public void Generate(ResultOutput output)
 
         foreach (var hook in webhooks)
         {
+            var names = new List<string> { hook.Metadata.SingularName, hook.Metadata.Group ?? string.Empty, hook.Metadata.Version };
+            var hookName = string.Join(".", names.Where(name => !string.IsNullOrWhiteSpace(name)));
             mutatorConfig.Webhooks.Add(new V1MutatingWebhook
             {
-                Name = $"mutate.{hook.Metadata.SingularName}.{hook.Metadata.Group}.{hook.Metadata.Version}",
+                Name = $"mutate.{hookName}",

```

Comment:
As far as I know, if no "group" is present, Kubernetes internal stuff defaults to "Core".

Shouldn't we default to "Core" if there is no group present? Either in the metadata class or in this logic

Reply:
It's a shame to let this fix languish, I got a way into building a validating webhook before finding it doesn't work with pods. Is there a workaround?

Reply:
I don't know. Never had the issue and never got a serious answer 🤷🏻

Reply:
Addition: If you have a good take on it, or an opinion, feel free to share.
I thought using "default" for the group is a promising idea since it is the default behavior of Kubernetes.